tools/wt_cmp_dir

```diff
@@ -0,0 +1,130 @@
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+# Compare content in WT home directories.
+
+usage_exit() {
+    echo "Usage: wt_cmp_dir [ -vq ] [ -t timestamp ] dir1 [ -t timestamp  ] dir2"
+    echo "   -v   verbose"
+    exit 1
+}
+
+wtfiles() {
+    local tmpa=/tmp/wcd$$a
+    local tmpb=/tmp/wcd$$b
+    lastwt=
+    wt=wt
+    while [ "$wt" != "$lastwt" -a ! -x "$wt" ]; do
+        wt=../$wt
+    done
+    #TODO: check for corruption
+    $wt -h "$1" list >$tmpa 2>$tmpb
+    if [ "$?" != 0 -o -s $tmpb ]; then
+        echo failed command: $wt -h "$1" list
+        cat $tmpb
+        rm -f $tmpa $tmpb
+        exit 1
+    fi
+    uris=$(grep -v colgroup: < $tmpa)
+    rm -f $tmpa $tmpb
+    #echo INITIAL URIS $uris >&2
+    results=""
+    for uri in $uris; do
+        case "$uri" in
+            file:* )
+                base=$(echo "$uri" | sed -e 's/file://' -e 's/.wt//')
+                if ! echo "$uris" | grep "table:$base" > /dev/null; then
+                    results="$results $uri"
+                fi
+            ;;
+            table:* )
+                results="$results $uri"
+                ;;
+        esac
+    done
+    #echo EDITED URIS $results >&2
+    echo $results
+}
+
+timestamp_opt1=
+timestamp_opt2=
+
+while [ "$#" -gt 2 ]; do
+    if [ "$1" = '-t' ]; then
+        # By default, the first -t argument, if given
+        # applies to both directories
+        timestamp_opt1="-t $2"
+        timestamp_opt2="-t $2"
+        shift; shift
+    else
+        break  # done with flags
+    fi
+done
+if [ "$#" -lt 2 ]; then
+    usage_exit
+fi
+
+# The next argument is the first directory
+dir1="$1"
+files1=$(wtfiles "$dir1")
+shift
+
+# We can still have another -t argument to apply to the
+# second directory.
+
+if [ "$#" -gt 2 -a "$1" = '-t' ]; then
+    timestamp_opt2="-t $2"
+    shift; shift
+fi
+
+if [ "$#" != 1 ]; then
+    usage_exit
+fi
+
+# The next argument is the second directory
+dir2="$1"
+files2=$(wtfiles "$dir2")
+
+# If the home directories have different files, give up.
+# In the future we might consider relaxing this, reporting missing files,
+# and doing the compare for common files, in the spirit of diff -r.
+
+if [ "$files1" != "$files2" ]; then
+    echo "Directories have different files to compare, stopping."
+    echo "  $dir1: $files1"
+    echo "  $dir2: $files2"
+    exit 1
+fi
+
+# Get the script to run
+cmp_uri_script=$(dirname "$0")/wt_cmp_uri
+
+for f in $files1; do
+    echo $f
+    python $cmp_uri_script $timestamp_opt1 "$dir1"/$f $timestamp_opt2 "$dir2"/$f
+done
```

Comment:
Should we exit with an error code if there was a difference in any of these files/tables?